internal/tools/aws/ec2_types.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+)
+
+// EC2API represents the series of calls we require from the AWS SDK v2 EC2 Client
+type EC2API interface {
```

Comment:
Because the v1 SDK provided interfaces for all services but the v2 does not, so we need to create our own with the methods we use so we can properly mock those in tests using `gomock`.

Reply:
Understood